### PR TITLE
Upgrade sailfish to 0.7.1.

### PR DESCRIPTION
**Pull request by roryk**
1) Tested and builds on OSX and Linux
2) Removed the keg-only designation, since it doesn't seem to interfere
with jellyfish anymore.
3) /usr/local/bin needs to be injected into the path to pick up autoconf.

### CODE REVIEW

**Comment by roryk** [comment]
I'm not sure why env.deparallelize is necessary but I left it in there because it was there already. Not picking up autoconf is strange, but I couldn't find a nicer way to do it other than adding the path.

File: sailfish.rb [hunk 1]
```diff
@@ -2,29 +2,21 @@ class Sailfish < Formula
...
-  end
-
-  fails_with :gcc do
-    build 5666
-    cause "Sailfish requires g++ 4.7 or greater."
-  end
-
   def install
     ENV.deparallelize
+    ENV.append "PATH", "/usr/local/bin", ":"
```
Comment (dunn) [inline sailfish.rb]: If it needs autotools, add these lines instead:
```ruby
depends_on "automake" => :build
depends_on "autoconf" => :build

Comment (roryk) [inline sailfish.rb]: Hi @dunn,

Thanks!

**Comment by roryk** [comment]
I don't understand why this is failing any tests, seems like it is working fine to me.

**Comment by dpo** [comment]
Seems to be the bot that's confused. I started off a new build.

**Comment by roryk** [comment]
Thanks @dpo, looks like something is still up.

**Comment by roryk** [comment]
I still don't get why the bot fails on this, looks totally fine when I run it. Anyone have any ideas?

**Comment by roryk** [comment]
Any ideas?

File: sailfish.rb [hunk 2]
```diff
@@ -2,29 +2,22 @@ class Sailfish < Formula
   homepage "http://www.cs.cmu.edu/~ckingsf/software/sailfish"
   # doi "10.1038/nbt.2862"
   # tag "bioinformatics"
-  url "https://github.com/kingsfordgroup/sailfish/archive/v0.6.3.tar.gz"
-  sha256 "5fded7dc88e525d17eb7abe102ac8ce8d8c3c8e9997d8aa1c512dddbef813632"
+  url "https://github.com/kingsfordgroup/sailfish/archive/v0.7.4.tar.gz"
+  sha256 "7ad5ee24b63611688a43617f2d5b48711b36014df0f20d418ec1f4c0a9375854"
+  version "0.7.4"
```
Comment (dpo) [inline sailfish.rb]: It's a simple matter of removing the `version` line now. Thanks and sorry for the wait!
```
==> brew audit Homebrew/science/sailfish				 FAILED
Error: 1 problem in 1 formula
homebrew/science/sailfish:
 * Stable: version 0.7.4 is redundant with version scanned from URL
```

**Comment by roryk** [comment]
Thanks @dpo,

No worries, sorry to be a pain. I dropped the redundant version line.

**Comment by dpo** [comment]
Build error on Mountain Lion:
```
/usr/local/Library/ENV/4.3/clang++    -pthread -funroll-loops -fPIC -fomit-frame-pointer -Ofast -DHAVE_ANSI_TERM -DHAVE_SSTREAM -DRAPMAP_SALMON_SUPPORT -Wall -std=c++11 -Wreturn-type -Werror=return-type -undefined dynamic_lookup -stdlib=libc++ -DBOOST_HAS_INT128 -Wno-deprecated-register -I/tmp/sailfish20150911-68894-1jxvoh/sailfish-0.7.4/include -I/tmp/sailfish20150911-68894-1jxvoh/sailfish-0.7.4/include/eigen3 -I/tmp/sailfish20150911-68894-1jxvoh/sailfish-0.7.4/external -I/tmp/sailfish20150911-68894-1jxvoh/sailfish-0.7.4/external/cereal/include -I/tmp/sailfish20150911-68894-1jxvoh/sailfish-0.7.4/external/install/include -I/tmp/sailfish20150911-68894-1jxvoh/sailfish-0.7.4/external/install/include/rapmap -I/tmp/sailfish20150911-68894-1jxvoh/sailfish-0.7.4/external/install/include/jellyfish-2.1.3 -I/usr/local/include    -o CMakeFiles/sailfish_core.dir/SailfishIndexer.cpp.o -c /tmp/sailfish20150911-68894-1jxvoh/sailfish-0.7.4/src/SailfishIndexer.cpp
In file included from /tmp/sailfish20150911-68894-1jxvoh/sailfish-0.7.4/src/SailfishIndexer.cpp:60:
In file included from /tmp/sailfish20150911-68894-1jxvoh/sailfish-0.7.4/external/install/include/rapmap/RapMapSAIndex.hpp:12:
In file included from /tmp/sailfish20150911-68894-1jxvoh/sailfish-0.7.4/external/install/include/google/dense_hash_map:34:
/tmp/sailfish20150911-68894-1jxvoh/sailfish-0.7.4/external/install/include/sparsehash/dense_hash_map:106:10: fatal error: 'tr1/functional' file not found
#include HASH_FUN_H                 // for hash<>
         ^
/tmp/sailfish20150911-68894-1jxvoh/sailfish-0.7.4/external/install/include/sparsehash/internal/sparseconfig.h:10:20: note: expanded from macro 'HASH_FUN_H'
#define HASH_FUN_H <tr1/functional>
                   ^
1 error generated.
```
Quite sure I've seen it before...

**Comment by dunn** [comment]
Adding `ENV.libcxx` to the install block fixed a similar error in #2265.

**Comment by roryk** [comment]
Thanks for the hand-holding @dpo and @dunn. I can't actually see what the errors are. :)

**Comment by dunn** [comment]
You should be able to click the "Details" link below where it says "All checks have failed".

**Comment by dunn** [comment]
The most recent failure was a test bot error, though. I re-queued the build.

**Comment by dunn** [comment]
Everything passed except the Mountain Lion test:
```
==> /usr/local/Cellar/sailfish/0.7.4/bin/sailfish --version
dyld: Symbol not found: __ZNK5boost15program_options22error_with_option_name23substitute_placeholdersERKNSt3__112basic_stringIcNS2_11char_traitsIcEENS2_9allocatorIcEEEE
  Referenced from: /usr/local/Cellar/sailfish/0.7.4/bin/sailfish
  Expected in: flat namespace
 in /usr/local/Cellar/sailfish/0.7.4/bin/sailfish
```

**Comment by roryk** [comment]
Thanks @dunn,

@sjackman-- I see you solved a similar issue with Salmon, but I don't see how you did that. Do you know how to fix this?

File: sailfish.rb [hunk 3]
```diff
@@ -2,29 +2,26 @@ class Sailfish < Formula
...
-
-  fails_with :clang do
-    cause "Currently, the only supported compiler is GCC(>=4.7). We hope to support Clang soon."
-  end
-
-  fails_with :gcc do
-    build 5666
-    cause "Sailfish requires g++ 4.7 or greater."
-  end
+  needs:cxx11
```
Comment (iml) [inline sailfish.rb]: Separate these `needs :cxx11`

File: sailfish.rb [hunk 4]
```diff
@@ -2,29 +2,26 @@ class Sailfish < Formula
...
+  # mountain lion build is broke and can't figure out how to fix it
+  depends_on :macos => :mavericks
 
   def install
     ENV.deparallelize
+    ENV.libcxx if MacOS.version < :mavericks
```
Comment (iml) [inline sailfish.rb]: You can remove this if we're ending support for <= Mountain Lion.

File: sailfish.rb [hunk 5]
```diff
@@ -2,29 +2,26 @@ class Sailfish < Formula
...
     mkdir "build" do
       system "cmake", "..", *std_cmake_args
-      system "make", "install"
+      system "make"
+      bin.install("src/sailfish")
```
Comment (iml) [inline sailfish.rb]: Why not make install?

**Comment by roryk** [comment]
Thanks for the suggestions for cleaning up for formula, @iml. I implemented those changes and tested and it works fine on my Ubuntu VM as well as Yosemite.

**Comment by roryk** [comment]
Thanks everyone, why is this still failing?

**Comment by dpo** [comment]
Bot glitch. I restarted the job.

**Comment by dpo** [comment]
Sorry to make you wait. There's just an audit warning on 10.9 (run `brew audit --strict sailfish`). But there's an error on 10.10:
```
/usr/local/Library/ENV/4.3/clang++ -DHAVE_CONFIG_H -I. -I./src  -I./src  -Wall -W -Wwrite-strings -Woverloaded-virtual -Wshadow -g -O2 -MT hashtable_test.o -MD -MP -MF .deps/hashtable_test.Tpo -c -o hashtable_test.o `test -f 'src/hashtable_test.cc' || echo './'`src/hashtable_test.cc
src/hashtable_test.cc:904:3: error: no matching function for call to 'swap'
  swap(this->ht_, other_ht);
  ^~~~
```

**Comment by roryk** [comment]
Sweet, looks like this fixed the issues on Mountain Lion as well.

**Comment by dpo** [comment]
Excellent! Thanks for all this. Would you mind squashing your commits together?

**Comment by roryk** [comment]
Thanks @dpo for the patience. All good.

File: sailfish.rb [hunk 6]
```diff
@@ -1,24 +1,17 @@
...
   homepage "http://www.cs.cmu.edu/~ckingsf/software/sailfish"
   # doi "10.1038/nbt.2862"
   # tag "bioinformatics"
-  url "https://github.com/kingsfordgroup/sailfish/archive/v0.6.3.tar.gz"
-  sha256 "5fded7dc88e525d17eb7abe102ac8ce8d8c3c8e9997d8aa1c512dddbef813632"
+  url "https://github.com/kingsfordgroup/sailfish/archive/v0.7.5.tar.gz"
+  sha256 "74f589f6dfa8d9316a9a77bb8ad9c2962f08e3a285deedc134893f6f7b1d1b18"
 
+  depends_on "automake" => :build
+  depends_on "autoconf" => :build
```
Comment (sjackman) [inline sailfish.rb]: Please list dependencies in alphabetical order.

**Comment by roryk** [comment]
Thanks @sjackman, fixed.

**Comment by sjackman** [comment]
Wow, that was fast. It's a minor point, but I needed to trigger a bot rebuild anyway. Thanks, Rory.

**Comment by sjackman** [comment]
You've been very patient.

**Comment by roryk** [comment]
No worries, you folks are busy.

**Comment by sjackman** [comment]
```
curl -I --silent --fail --output /dev/null https://homebrew.bintray.com/bottles-science/sailfish-0.7.5.el_capitan.bottle.tar.gz
Error: sailfish-0.7.5.el_capitan.bottle.tar.gz is already published. Please remove it manually from
https://bintray.com/homebrew/bottles-science/sailfish/view#files
Build step 'Execute shell' marked build as failure
Started calculate disk usage of build
Finished Calculation of disk usage of build in 0 seconds
Started calculate disk usage of workspace
Finished Calculation of disk usage of workspace in  1 second
[WS-CLEANUP] Deleting project workspace...[WS-CLEANUP] done
Finished: FAILURE
```
http://bot.brew.sh/job/Homebrew%20Bottles/33759/console

**Comment by sjackman** [comment]
Uploading the bottles to Bintray failed because version 0.7.5 bottles had already been uploaded *and published*. I deleted the offending bottles. I also learned that you can rebuild just the downstream CI stage of building and uploading the bottles without having to rebuild the package itself. That saves a lot of time.

**Comment by sjackman** [comment]
Thanks for your contribution, Rory. I'm glad that we finally got it merged.

**Comment by sjackman** [comment]
The first line of the commit log message should be simply `sailfish 0.7.5`

**Comment by roryk** [comment]
Thanks @sjackman!

**Pull request closed without merging**